src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithNullTest.java

```diff
@@ -110,6 +110,45 @@ public void testAddbatch2AEOnConnection() throws SQLException {
     public void testAddbatch2() throws SQLException {
         testAddBatch2(getConnection());
     }
+    
+    @Test
+    public void testClearBatch() throws SQLException {
+        String batchTable = RandomUtil.getIdentifier("batchTable");
+        String CREATE_TABLE_SQL = "create table " + batchTable + " (KEY1 numeric(19,0) not null, KEY2 numeric(19,0) not null, primary key (KEY1, KEY2))";
+        String INSERT_ROW_SQL = "INSERT INTO " + batchTable + "(KEY1, KEY2) VALUES(?, ?)";
+        
+        try (Connection conn = getConnection(); Statement s = conn.createStatement()) {
+            try (PreparedStatement pstmt = conn.prepareStatement(INSERT_ROW_SQL)) {
+                s.execute(CREATE_TABLE_SQL);
+                // Set auto-commit to false
+                conn.setAutoCommit(false);
+                executeBatch(pstmt, 10, "foo".hashCode() + 1);
+                pstmt.clearParameters();
+                executeBatch(pstmt, 10, "bar".hashCode() + 2);
+                conn.commit();
+            } catch (Exception e) {
+                conn.rollback();
+                throw e;
+            } finally {
+                TestUtils.dropTableIfExists(batchTable, s);
+            }
+        }
+    }
+
+    private void executeBatch(PreparedStatement pstmt, int count, int run) throws SQLException {
+        long base = System.currentTimeMillis();
+
+        for (int idx = 1; idx <= count; idx++) {
+            pstmt.setLong(1, base + idx);
+            pstmt.setLong(2, run);
+            pstmt.addBatch();
+        }
+
+        int[] rowCounts = pstmt.executeBatch();
+        for (int idx = 0; idx < rowCounts.length; idx++) {
+            System.out.printf("Row %2d %s successfully inserted.\n", idx + 1, rowCounts[idx] == 1 ? "was" : "was not");
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.